index.bs

```diff
@@ -575,68 +575,182 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+			1. Let |p| be [=a new promise=] in |realm|.
+
+			1. Let |blobData| be a [=Blob=] created using [=representation=]'s [=data=] corresponding to the |mimeType|.
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			1. Resolve |p| with |blobData|.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+			1. Return |p|.
+
+		<dl class=note>
+			<dt><code><var>clipboardItem</var> = new ClipboardItem([<var>items</var>, <var>options</var>])</code>
+			<dd>
+			Creates a new {{ClipboardItem}} object. <var>items</var> are used to fill its MIME types and [=Promise=]s to [=Blob=]s or {{DOMString}}s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
+
+				<pre class="example javascript" highlight=js>
+					const format1 = 'text/plain';
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{presentationStyle: "unspecified"});
+				</pre>
+
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
+
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		A <dfn>Clipboard</dfn> may contain <dfn>ClipboardItems</dfn> which has one or more [=clipboard item=]s. Multiple [=clipboard item=]s are only supported on platforms such as iOS/iPadOS.
+		A web author needs to create a |data| which is a [=ClipboardItems=] object in order to write content to clipboard using the {{Clipboard/write(data)}} method.
+		{{Clipboard/read()}} returns a [=Promise=] to [=ClipboardItems=] object that represents contents of clipboard.
+		A [=clipboard item=] can be read from [=ClipboardItems=].
 
+		<div id="clipboard-idl" dfn-for="Clipboard">
 		<div class="algorithm" data-algorithm="clipboard-read">
 		<h4 method for="Clipboard">read()</h4>
 		The {{Clipboard/read()}} method must run these steps:
 
-			1. Let |p| be a new [=Promise=].
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |p| be [=a new promise=] in |realm|.
 
 			1. Run the following steps [=in parallel=]:
 
 				1. Let |r| be the result of running [=check clipboard read permission=] [=in parallel=]
```

Comment:
```suggestion
				1. Let |r| be the result of running [=check clipboard read permission=]
```

(we're already "in parallel" so this is not necessary).

Reply:
Done.

Reply:
Ack.

Reply:
Ack.